proposals/0010-resource-element-type-validation.md

```diff
@@ -0,0 +1,217 @@
+* Proposal: [0010](0010-resource-element-type-validation.md)
+* Author(s): [Joshua Batista](https://github.com/bob80905)
+* Sponsor: Joshua Batista
+* Status: **Under Consideration**
+* Impacted Project(s): (LLVM)
+* Issues: [#75676](https://github.com/llvm/llvm-project/issues/75676)
+
+## Introduction
+Resources are often used in HLSL, with various resource element types.
+
+For example:
+```
+RWBuffer<float> rwbuf: register(u0);
+```
+In this code, the element type is `float`, and the resource type is `RWBuffer`.
+`RWBuffer`, along with some other buffers and textures, fall under the "typed buffer"
+category. Other buffer resources are categorized as "raw buffers". 
+Below is a description of resources that are considered "typed buffers" vs "raw buffers".
+* typed buffers
+  * [RW|RasterizerOrdered]Buffer
+  * [Feedback]Texture*
+* raw buffers
+  * [Append|Consume|RW|RasterizerOrdered]StructuredBuffer
+  * [RW]ByteAddressBuffer
+
+There is a distinct set of rules that define valid element types for typed buffer resources
+and valid element types for raw buffer resources.

```

Comment:
Nit: I think the language of "element types" applied to "raw buffer resources" is a little problematic.  StructuredBuffer has an element (or struct) type.  ByteAddressBuffer does not have an "element type", but has templated Load/Store operations that take a type.

Perhaps being more explicit could just be the answer:
```suggestion
and valid element types for `*StructuredBuffer` resources and
`*ByteAddressBuffer` Load/Store operations.
```